src/SessionManager.cc

```diff
@@ -356,117 +323,62 @@ Connection* NetSessions::FindConnection(Val* v)
 	id.dst_port = htons((unsigned short) resp_portv->Port());
 
 	id.is_one_way = false;	// ### incorrect for ICMP connections
+	id.proto = orig_portv->PortType();
 
-	detail::ConnIDKey key = detail::BuildConnIDKey(id);
-	ConnectionMap* d;
-
-	if ( orig_portv->IsTCP() )
-		d = &tcp_conns;
-	else if ( orig_portv->IsUDP() )
-		d = &udp_conns;
-	else if ( orig_portv->IsICMP() )
-		d = &icmp_conns;
-	else
-		{
-		// This can happen due to pseudo-connections we
-		// construct, for example for packet headers embedded
-		// in ICMPs.
-		return nullptr;
-		}
+	detail::ConnIDKey conn_key = detail::BuildConnIDKey(id);
+	detail::SessionKey key(&conn_key, sizeof(conn_key), false);
 
 	Connection* conn = nullptr;
-	auto it = d->find(key);
-	if ( it != d->end() )
-		conn = it->second;
+	auto it = session_map.find(key);
+	if ( it != session_map.end() )
+		conn = static_cast<Connection*>(it->second);
 
 	return conn;
 	}
 
-void NetSessions::Remove(Connection* c)
+void SessionManager::Remove(Session* s)
 	{
-	if ( c->IsKeyValid() )
+	Connection* c = static_cast<Connection*>(s);
+
+	if ( s->IsKeyValid() )
 		{
-		const detail::ConnIDKey& key = c->Key();
-		c->CancelTimers();
+		s->CancelTimers();
+		s->Done();
+		s->RemovalEvent();
 
-		if ( c->ConnTransport() == TRANSPORT_TCP )
-			{
-			auto ta = static_cast<analyzer::tcp::TCP_Analyzer*>(c->GetRootAnalyzer());
-			assert(ta->IsAnalyzer("TCP"));
-			analyzer::tcp::TCP_Endpoint* to = ta->Orig();
-			analyzer::tcp::TCP_Endpoint* tr = ta->Resp();
+		// Clears out the session's copy of the key so that if the
+		// session has been Ref()'d somewhere, we know that on a future
+		// call to Remove() that it's no longer in the map.
```

Comment:
Should this comment stay with the `ClearKey` below?

Reply:
Yep, I missed moving it when I moved `ClearKey` during my rebasing yesterday.